subsys/net/lib/coap/coap_server.c

```diff
@@ -136,9 +136,8 @@ static int coap_server_process(int sock_fd)
 	ssize_t received;
 	int ret;
 
-	received = zsock_recvfrom(sock_fd, buf, sizeof(buf), ZSOCK_MSG_DONTWAIT, &client_addr,
-				  &client_addr_len);
-	__ASSERT_NO_MSG(received <= sizeof(buf));
+	received = zsock_recvfrom(sock_fd, buf, sizeof(buf), ZSOCK_MSG_DONTWAIT | ZSOCK_MSG_TRUNC,
```

Comment:
Could be make the MSG_TRUNC usage optional (can be enabled by default) as in case of coap client (
https://github.com/zephyrproject-rtos/zephyr/commit/715b97397b797c7f105053dd41f2532006ae909d)? We had some issues with that flag in the offloaded case.